astropy/coordinates/baseframe.py

```diff
@@ -475,6 +475,28 @@ def realize_frame(self, representation):
                         if attr not in self._attr_names_with_defaults])
         return self.__class__(representation, **frattrs)
 
+    def clone_frame(self, **kwargs):
+        """
+        Generates a new frame *without data* from this frame (which may or
+        may not have data).  Optinally, can override frame attributes.
+        Roughly speaking, the converse of `realize_frame`.
+
+        Parameters
+        ----------
+        Any keywords are treated as frame attributes to be set on the new frame
+        object.
+
+        Returns
+        -------
+        frameobj : same as this frame
+            A new object with the same frame attributes as this one, but
+            with no data and possibly new frame attributes.
+        """
+        attrs_to_set = {nm:getattr(self, nm) for nm in self.frame_attributes}
```

Comment:
This should exclude defaults (i.e., just copy `frattrs` definition from `realize_frame`)